src/client/services/cdc/index.js

```diff
@@ -2,26 +2,14 @@ const io = require('socket.io-client');
 let socket = io.connect('/');
 
 const CDC = {
-  initGraphSocket(updateFunction) {
-    socket.on('layoutPackage', (cyZip) => {
-      let msg = atob(cyZip);
-      try {
-        let cyJSON = JSON.parse(msg);
-        updateFunction(cyJSON);
-      } catch(err) {
-        if (msg.length > 0) {
-          console.log(JSON.parse(msg));
-          //console.log(msg);
-        } else {
-          throw err;
-        }
+  getGraphAndLayout(uri, version) {
+    return fetch(`http://${window.location.hostname}/get-graph-and-layout?uri=${encodeURIComponent(uri)}&version=${version}`, {
```

Comment:
Does the following code not work?
```js
fetch(`/get-graph-and-layout?uri=${encodeURIComponent(uri)}&version=${version}`, ... );
```

Reply:
You should not be using window at all when possible unless you like programming with global variables.

Reply:
I'll change it to that. It works. Thanks for the tip!